stl/inc/algorithm

```diff
@@ -10102,9 +10102,17 @@ namespace ranges {
             if constexpr (forward_range<_Rng> && _Prefer_iterator_copies<iterator_t<_Rng>>) {
                 const auto _Found = _RANGES _Minmax_element_unchecked(
                     _STD move(_UFirst), _STD move(_ULast), _Pass_fn(_Pred), _Pass_fn(_Proj));
+                // We need to detect move_iterators (and similar types) to avoid dereferencing them more than once.
+                // The `if constexpr` is a pure optimization; we don't want an additional branch for normal iterators.
+                if constexpr (!is_lvalue_reference_v<range_reference_t<_Rng>>) {

```

Comment:
"Is not an lvalue reference" isn't a good choice, IMO - I can't prove it's always correct. The iterator could dereference to an lvalue of a proxy reference for which construction of the range value type causes element modification, for example. I'd prefer that we avoid the branch only when the range's reference type is cv-`range_value_t` or an lvalue reference to such.